### ISSUE TITLE
Output as wav

### ISSUE THREAD

**Issue by audiofool** [HEnquist/camilladsp#193]
I am having some success using camilladsp within LMS(slimserver) as part of my DSP upsampling chain.  I pipe to camilladsp which then pipes to stdout.  I currently have to add sox to the chain to convert from raw to wav format which LMS(squeezelite) requires.  Is it possible to add a wav output option?

**Comment by HEnquist** [comment]
Yes this would be quite straightforward. There is only one tricky part. The wav header contains the total length of the file, which isn't known when the header is written. It should be fine to set the length to the maximum possible value. And if that causes trouble then it would be possible to rewind and update the length before closing the file.

**Comment by audiofool** [comment]
In my current process with sox I include --ignore-length since I am not writing to a file but using stdout to squeezelite which I believe ignores length, so setting to max value should work.

**Comment by Foxenfurter** [comment]
Hi - I am continuing with my plugin work on LMS using CamillaDSP as discussed in another Issue. I have a basic implementation running with core features added and I am now trying to iron out the wrinkles (it is very wrinkly!). Like audiofool, I really need a wav output so I was wondering whether this feature is available?

I will assume it isn't and will add in the extra sox command for now.

**Comment by HEnquist** [comment]
It's not done yet. I'll try to include it in the next version.

**Comment by Foxenfurter** [comment]
Thanks - no rush. I managed to add in a sox conversion within my embedded app and it seems to be working OK.

**Comment by HEnquist** [comment]
implemented in https://github.com/HEnquist/camilladsp/pull/325
